### PR TITLE
Feature/end cautionary alert

### PR DESCRIPTION
**Pull request by humulla**
Add EndCautionaryAlert object with validator tests and factory methods

### CODE REVIEW

File: Hackney.Shared.CautionaryAlerts/Boundary/Request/Validation/EndCautionaryAlertValidator.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+            RuleFor(x => x.Alert).SetValidator(new AlertValidator());
+
+            RuleFor(x => x.AssetDetails).SetValidator(new AssetDetailsValidator());
+
+            RuleFor(x => x.PersonDetails).SetValidator(new PersonDetailsValidator());
+
+            RuleFor(x => x.AlertId).NotEmpty().NotNull();
+
+            RuleFor(x => x.IsActive).Must(x => x.Equals(false));
```
Comment (FaisalHackney) [inline Hackney.Shared.CautionaryAlerts/Boundary/Request/Validation/EndCautionaryAlertValidator.cs:37]: Question: does the IsActive not get validated as true at initial stage?

File: Hackney.Shared.CautionaryAlerts.Tests/Boundary/Validation/EndCautionaryAlertValidatorTests.cs [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+                                        .Without(x => x.AlertId)
+                                        .Create();
+
+            var result = _classUnderTest.TestValidate(cautionaryAlert);
+            result.ShouldHaveValidationErrorFor(x => x.AlertId);
+
+        }
+
+        [Test]
+        public void AlertIdShouldFailIfEmpoty()
```
Comment (FaisalHackney) [inline Hackney.Shared.CautionaryAlerts.Tests/Boundary/Validation/EndCautionaryAlertValidatorTests.cs]: Spelling mistake in Test name

File: Hackney.Shared.CautionaryAlerts/Boundary/Request/Validation/EndCautionaryAlertValidator.cs (hunk 1 above)
Comment (humulla) [reply Hackney.Shared.CautionaryAlerts/Boundary/Request/Validation/EndCautionaryAlertValidator.cs:37]: Yes, but when you end a cautionary alert the request should send through a request with isActive set to false

Comment (FaisalHackney) [reply Hackney.Shared.CautionaryAlerts/Boundary/Request/Validation/EndCautionaryAlertValidator.cs:37]: ah ok

**Review by FaisalHackney — APPROVED** [review]

**Pull request merged**